metau-capstone/StudyViewController.m

```diff
@@ -10,14 +10,79 @@
 #import "SceneDelegate.h"
 #import "LoginViewController.h"
 @interface StudyViewController ()
-
+@property (nonatomic, strong) CALayer *front;
+@property (nonatomic, strong) CALayer *back;
+@property (nonatomic, strong) CABasicAnimation *rotateAnim;
+@property (nonatomic) CATransform3D horizontalFlip;
+@property (nonatomic) BOOL isFlipped;
 @end
 
 @implementation StudyViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    // Do any additional setup after loading the view.
+    //BACK SIDE
+    self.back = [[CALayer alloc] init];
+    self.back.frame = CGRectMake(0, 0, 300, 180);
+    self.back.backgroundColor = [[UIColor blackColor] CGColor];
+    self.back.position = CGPointMake(self.view.center.x, self.view.center.y - 50);
+    
+    // add text label to the flashcard
+    CATextLayer *backLabel = [[CATextLayer alloc] init];
+    [backLabel setFont:@"Helvetica-Bold"];
+    [backLabel setFontSize:20];
+    [backLabel setString:@"Back"];
+    [backLabel setAlignmentMode:kCAAlignmentCenter];
+    [backLabel setForegroundColor:[[UIColor whiteColor] CGColor]];
+    [backLabel setFrame:CGRectMake(0, 0, 300, 180)];
+    [self.back addSublayer:backLabel];
+    self.back.transform = CATransform3DMakeRotation(M_PI, 0, -1, 0);
+    [self.view.layer addSublayer:self.back];
+    
+    // FRONT SIDE
+    self.front = [[CALayer alloc] init];
+    self.front.frame = CGRectMake(0, 0, 300, 180);
+    self.front.backgroundColor = [[UIColor whiteColor] CGColor];
+    self.front.position = CGPointMake(self.view.center.x, self.view.center.y - 50);
+    
+    // add text label to the flashcard
+    CATextLayer *label = [[CATextLayer alloc] init];
+    [label setFont:@"Helvetica-Bold"];
+    [label setFontSize:20];
+    [label setString:@"Front"];
+    [label setAlignmentMode:kCAAlignmentCenter];
+    [label setForegroundColor:[[UIColor blackColor] CGColor]];
+    [label setFrame:CGRectMake(0, 0, 300, 180)];
+    [self.front addSublayer:label];
+    
+    [self.view.layer addSublayer:self.front];
+    
+    // create rotation animation
+    self.rotateAnim = [CABasicAnimation animationWithKeyPath:@"transform.rotation.y"];
```

Comment:
I noticed some flickering during the flip animation. Have you tried using UIView.transition function instead?
example in swift: 
https://github.com/Sajjon/SwiftIntro/blob/master/SwiftIntro/Views/Cells/CollectionViewCells/CardCVCell/CardCVCell.swift